gitpod-io/gitpod
# [supervisor] Harmonize and future-proof API

supervisor is the root process in a workspace container. It's the one that starts the actuall IDE (Theia), but also prepares the environment, provides a health endpointm and offers services for full-workspace backup.

Over time the duties of supervisor have grown and we're just short of the next step. In preparation, we must harmonize the "health endpoint" and make it ready for future tasks. To this end, this PR:
- renames the health endpoint to API endpoint
- harmonizes the API and introduces the `/_supervisor` route which provides access to said API. Some calls don't require authentication, most do.

This PR also prepares moving some conerns out of Theia, e.g. IDE health status. Until now the `/gitpod/ready` endpoint was used to determine if the IDE was running. This required modification of the IDE itself. With this PR we'll have a `/_supervisor/v1/status/ide` endpoint which will be maintained by supervisor (follow up PR).

## How to test
- start a workspace - if it starts most is already well
- make a call to the supervisor API:
  ```
  curl localhost:22999/_supervisor/v1/status/ide
  curl $(gp url)/_supervisor/v1/status/ide
  ```
- make a call to supervisor API which requires authentication (should fail with 401):
  ```
  curl -v $(gp url)/_supervisor/v1/status/backup
  ```
